File: AppAcademySkypeInterviewPractice/OrderedVowelWords.py
test_string = 'this is a test of the vowel ordering system'

def OrderedVowelWords(test_string):
	ans = []
	words = test_string.split(' ')
	for word in words:
		if InOrder(word):
			ans.append(word)
	return(' '.join(ans))

def InOrder(word):
	vowels_present = []
	vowels = 'aeiou'
	for let in word:
		if let in vowels:
			vowels_present.append(let)
	if sorted(vowels_present) != vowels_present:
		return(False)
	#for let in range(len(vowels_present)-1):
		#if vowels_present[let] > vowels_present[let+1]:
		#	return(False)
	return(True)

print(OrderedVowelWords(test_string))

File: AppAcademySkypeInterviewRuby/ordered_vowel_words.rb
def ordered_vowel_words(str)
	words = str.split(" ")
	good_words = []
	words.each do |word|
		good_words << word if ordered_vowel_word?(word)
	end
	good_words.join(" ")
end

def ordered_vowel_word?(str)
	vowels = %w[a e i o u]
	str_split = str.split("")
	vow = []
	str_split.each do |let|
		vow << let.ord if vowels.include? let
	end

	return true if vow == vow.sort

end

p ordered_vowel_words("this is a test of the vowel ordering system")